manim/mobject/mobject.py

```diff
@@ -425,13 +425,34 @@ def add(self, *mobjects: "Mobject"):
         self.submobjects = list_update(self.submobjects, mobjects)
         return self
 
+    def insert(self, index: int, mobject: Mobject):
+        """Inserts a mobject at a specific position into self.submobjects
+
+        Effectively just calls  ``self.submobjects.insert(index, mobject)``,
+        where ``self.submobjects`` is a list.
+
+        Highly adapted from ``Mobject.add``.
+
+        Parameters
+        ----------
+        index
+            The index at which
+        mobject
+            The mobject to be inserted.
+        """
+        if not isinstance(mobject, Mobject):
+            raise TypeError("All submobjects must be of type Mobject")
+        if mobject is self:
+            raise ValueError("Mobject cannot contain self")
+        self.submobjects.insert(index, mobject)
+
     def __add__(self, mobject):
         raise NotImplementedError
 
     def __iadd__(self, mobject):
         raise NotImplementedError
 
-    def add_to_back(self, *mobjects: "Mobject"):
+    def add_to_back(self, *mobjects: Mobject):
```

Comment:
I mean im all for typehint changes. But they don't really seem to be related to this PR 🤔 can we just move them to a different PR instead?
@behackl We should also setup a typeguide in general or discuss this in the team which kind of typing we want to use. Because else it's just a mess.

Reply:
- #2765

Reply:
It's been a while since this PR, but I think most of these changes are because of adding `from __future__ import annotations` at the top, which removes the need for quotes around certain types and allows the use of builtins. There's a good chance they were automated 🤔.

They aren't really relevant but I don't think they hurt to have. I'd be willing to move these to another PR for the sake of a cleaner history though.